src/AbstractMenu.js

```diff
@@ -131,50 +131,77 @@ export default class AbstractMenu extends Component {
             return i === currentIndex ? null : i;
         }
 
-        const currentIndex = children.indexOf(selectedItem);
+        const currentIndex = selectedItem ? selectedItem.index : -1;
         const nextEnabledChildIndex = findNextEnabledChildIndex(currentIndex);
 
         if (nextEnabledChildIndex !== null) {
             this.setState({
-                selectedItem: children[nextEnabledChildIndex],
+                selectedItem: {
+                    index: nextEnabledChildIndex,
+                    // We need to know the type of the selected item, so we can
+                    // check it during render and tryToOpenSubMenu.
+                    type: children[nextEnabledChildIndex].type
```

Comment:
We need to keep both index and the type of the element because we have to check if it's a sub menu etc. in two places :/